src/Botan.php

```diff
@@ -20,82 +23,107 @@
+     *  - block tracking from inside commands by setting the value to non-existent command
+     *  - override which command is tracked when commands call other commands with executedCommand()
+     *
+     * @var string
      */
     public static $command = '';
 
     /**
      * Initialize Botan
      *
-     * @param $token
+     * @param  string $token
+     * @param  integer $timeout
+     *
      * @throws \Longman\TelegramBot\Exception\TelegramException
      */
-    public static function initializeBotan($token)
+    public static function initializeBotan($token, $timeout = 3)
     {
-        if (empty($token) || !is_string($token)) {
-            throw new TelegramException('Botan token should be a string!');
+        if (empty($token)) {
+            throw new TelegramException('Botan token is empty!');
         }
-        self::$token = $token;
+
+        if (!is_numeric($timeout)) {
+            throw new TelegramException('Timeout must be a number!');
+        }
+
+        self::$token   = $token;
+        self::$client  = new Client(['base_uri' => self::$api_base_uri, 'timeout' => $timeout]);
+
         BotanDB::initializeBotanDb();
     }
 
     /**
-     * Lock function to make sure only the first command is reported
-     * ( in case commands are calling other commands $telegram->executedCommand() )
+     * Lock function to make sure only the first command is reported (the one user requested)
      *
-     * @param  string $command
+     *  This is in case commands are calling other commands with executedCommand()
+     *
+     * @param string $command
      */
     public static function lock($command = '')
     {
         if (empty(self::$command)) {
-            self::$command = $command;
+            self::$command = strtolower($command);
         }
     }
 
     /**
      * Track function
      *
-     * @param  string $input
+     * @param  \Longman\TelegramBot\Entities\Update $update
      * @param  string $command
      *
      * @return bool|string
      * @throws \Longman\TelegramBot\Exception\TelegramException
      */
-    public static function track($input, $command = '')
+    public static function track(Update $update, $command = '')
     {
-        if (empty(self::$token) || $command !== self::$command) {
+        if (empty(self::$token) || strtolower($command) !== self::$command) {
             return false;
         }
 
-        if (empty($input)) {
-            throw new TelegramException('Input is empty!');
+        if (empty($update)) {
+            throw new TelegramException('Update object is empty!');
         }
 
+        // Release the lock in case someone runs getUpdates in foreach loop
         self::$command = '';
 
-        $obj  = json_decode($input, true);
+        // For now, this is the only way
+        $update_data = (array) $update;
+
         $data = [];
-        if (isset($obj['message'])) {
-            $data       = $obj['message'];
+        if ($update->getMessage()) {
```

Comment:
Would be nice if we could find a way to shorten this epic `if else` block and make it more dynamic.
Not priority though, more a though since you've touched code here 😃